pkg/ccl/crosscluster/logical/logical_replication_job_test.go

```diff
@@ -1456,3 +1456,87 @@ func TestShowLogicalReplicationJobs(t *testing.T) {
+
+	// Create user with no privileges
+	dbA.Exec(t, fmt.Sprintf("CREATE USER %s", username.TestUser))
+	testuser := sqlutils.MakeSQLRunner(s.SQLConn(t, serverutils.User(username.TestUser), serverutils.DBName("a")))
+
+	t.Run("Viewjob test", func(t *testing.T) {
+		// NEED VIEWJOB system grant to view admin LDR jobs
+		result := testuser.QueryStr(t, "select * from [SHOW JOBS] where job_id=$1", jobAID)
+		require.Empty(t, result, "The user should see no rows without the VIEWJOB grant")
+
+		result = testuser.QueryStr(t, "select * from [SHOW LOGICAL REPLICATION JOBS] where job_id=$1", jobAID)
+		require.Empty(t, result, "The user should see no rows without the VIEWJOB grant")
+
+		var returnedJobID jobspb.JobID
+		dbA.Exec(t, fmt.Sprintf("GRANT SYSTEM VIEWJOB to %s", username.TestUser))
+		testuser.QueryRow(t, "select job_id from [SHOW JOBS] where job_id=$1", jobAID).Scan(&returnedJobID)
+		require.Equal(t, returnedJobID, jobAID, "The user should see the LDR job with the VIEWJOB grant")
+
+		testuser.QueryRow(t, "select job_id from [SHOW LOGICAL REPLICATION JOBS] where job_id=$1", jobAID).Scan(&returnedJobID)
+		require.Equal(t, returnedJobID, jobAID, "The user should see the LDR job with the VIEWJOB grant")
+	})
+
+	// Kill replication job so we can create one with the testuser
+	dbA.Exec(t, "CANCEL JOB $1", jobAID)
+	jobutils.WaitForJobToCancel(t, dbA, jobAID)
+
+	t.Run("Schema test", func(t *testing.T) {
+		dbA.Exec(t, "CREATE SCHEMA testschema")
+
+		// Ensure the testuser has CREATE privileges
+		testuser.ExpectErr(t, "user testuser does not have CREATE privilege on schema testschema", fmt.Sprintf(testingUDFAcceptProposedBaseWithSchema, "testschema", "tab"))
+		dbA.Exec(t, "GRANT CREATE ON SCHEMA testschema TO testuser")
+		testuser.Exec(t, fmt.Sprintf(testingUDFAcceptProposedBaseWithSchema, "testschema", "tab"))
+	})
+
+	t.Run("Replication grant test", func(t *testing.T) {
+		// Only start replication with the REPLICATION grant
+		testuser.ExpectErr(t, "user testuser does not have REPLICATION system privilege", "CREATE LOGICAL REPLICATION STREAM FROM TABLE tab ON $1 INTO TABLE tab WITH DEFAULT FUNCTION = 'testschema.repl_apply'", dbBURL.String())
+		dbA.Exec(t, fmt.Sprintf("GRANT SYSTEM REPLICATION TO %s", username.TestUser))
+		testuser.QueryRow(t, "CREATE LOGICAL REPLICATION STREAM FROM TABLE tab ON $1 INTO TABLE tab WITH DEFAULT FUNCTION = 'testschema.repl_apply'", dbBURL.String()).Scan(&jobAID)
+	})
+
+	t.Run("CONTROLJOB test", func(t *testing.T) {
+		// Attempt to pause job without CONTROLJOB
+		testuser.ExpectErr(t, fmt.Sprintf("user testuser does not have privileges for job %s", jobAID.String()), "PAUSE JOB $1", jobAID.String())
+
+		// GRANT CONTROLJOB
+		dbA.Exec(t, fmt.Sprintf("GRANT SYSTEM CONTROLJOB to %s", username.TestUser))
+
+		testuser.Exec(t, "PAUSE JOB $1", jobAID)
+		jobutils.WaitForJobToPause(t, dbA, jobAID)
+
+		testuser.Exec(t, "RESUME job $1", jobAID)
```

Comment:
nit: resuming the job is unnecessary.

Reply:
Ah needed this when I had the DLQ stuff underneath. Fixing